File: app/src/main/java/me/zheteng/android/videosaver/parser/provider/VimeoProvider.java
package me.zheteng.android.videosaver.parser.provider;

import me.zheteng.android.videosaver.parser.UrlProvider;

/**
 *
 */
public class VimeoProvider implements UrlProvider{
    private static final String VIMEO_CONFIG_URL = "https://player.vimeo.com/video/";
    private String mId;

    public VimeoProvider(String id) {
        mId = id;
    }

    public String getId() {
        return mId;
    }

    @Override
    public String getTargetUrl() {
        return VIMEO_CONFIG_URL + mId + "/config";
    }
}

File: app/src/main/kotlin/me/zheteng/android/videosaver/parse/provider/VimeoProvider.kt
package me.zheteng.android.videosaver.parse.provider

import me.zheteng.android.videosaver.parse.UrlProvider


/**

 */
class VimeoProvider(val id: String) : UrlProvider {

    override fun getTargetUrl(): String {
        return VIMEO_CONFIG_URL + id + "/config"
    }

    companion object {
        private val VIMEO_CONFIG_URL = "https://player.vimeo.com/video/"
    }
}
